kotlin-preprocessors/buildSrc/src/main/kotlin/publish/mavencentral/PublishToMavenCentralPlugin.kt

```diff
@@ -1,94 +1,119 @@
 
         target.afterEvaluate {
-            val ossrhUser = target.propOrEnv("GODOT_KOTLIN_MAVEN_CENTRAL_TOKEN_USERNAME")
-            val ossrhPassword = target.propOrEnv("GODOT_KOTLIN_MAVEN_CENTRAL_TOKEN_PASSWORD")
-            val signingKey = target.propOrEnv("GODOT_KOTLIN_GPG_PRIVATE_KEY_ASCII")
-            val signingPassword = target.propOrEnv("GODOT_KOTLIN_GPG_KEY_PASSPHRASE")
+            val mavenCentralUser = target.propOrEnv("ORG_GRADLE_PROJECT_mavenCentralUsername") ?: target.propOrEnv("mavenCentralUsername")
+            val mavenCentralPassword = target.propOrEnv("ORG_GRADLE_PROJECT_mavenCentralPassword") ?: target.propOrEnv("mavenCentralPassword")
+            val gpgInMemoryKey = target.propOrEnv("ORG_GRADLE_PROJECT_signingInMemoryKey") ?: target.propOrEnv("signingInMemoryKey")
+            val gpgPassword = target.propOrEnv("ORG_GRADLE_PROJECT_signingInMemoryKeyPassword") ?: target.propOrEnv("signingInMemoryKeyPassword")
 
-            val releaseMode = !(target.version as String).endsWith("-SNAPSHOT")
+            val canSign = mavenCentralUser != null && mavenCentralPassword != null && gpgInMemoryKey != null && gpgPassword != null
 
-            target.extensions.configure(SigningExtension::class.java) {
-                @Suppress("UnstableApiUsage")
-                useInMemoryPgpKeys(signingKey, signingPassword)
-                target.extensions.findByType(PublishingExtension::class.java)?.publications?.all {
-                    sign(this)
-                }
-            }
-
-            target.extensions.configure(PublishingExtension::class.java) {
-                repositories {
-                    maven {
-                        val targetRepo = if (releaseMode) {
-                            "https://s01.oss.sonatype.org/service/local/staging/deploy/maven2/"
-                        } else {
-                            "https://s01.oss.sonatype.org/content/repositories/snapshots/"
-                        }
-                        setUrl(targetRepo)
-
-                        credentials {
-                            username = ossrhUser
-                            password = ossrhPassword
-                        }
-                    }
-                }
+            target.extensions.getByType(PublishingExtension::class.java).apply {
                 publications {
                     all {
                         if (this is MavenPublication) {
+                            groupId = "com.utopia-rise"
+                            artifactId = if (artifactId.isNullOrEmpty()) target.name else artifactId
+                            version = target.version as String
+
                             pom {
-                                name.set("kotlin-preprocessors")
-                                description.set("Gradle plugin to define preprocessors for kotlin language")
+                                url.set("https://github.com/utopia-rise/godot-kotlin-jvm.git")
```

Comment:
the url is copy pasted from the kotlin jvm project and should be updated to match this project again